### ISSUE TITLE
Downloaded + Unread badge are inaccurate

### ISSUE THREAD

**Issue by jolo283** [inorichi/tachiyomi#2172]
**DO NOT OPEN ISSUES/REQUESTS RELATING TO EXTENSIONS/CATALOGUES IN THIS REPOSITORY. Open them at the following repository  https://github.com/inorichi/tachiyomi-extensions/**

**For all other requests Please fill out the form below and remove the first 3 lines of this template**

**App version:**  0.8.4

**Android version:** 9 (Samsung S10e)

**Issue/Request:** Downloaded + Unread badge are inaccurate

**Steps to reproduce (if applicable)**

I noticed that when downloading a large number of chapters, the unread and downloaded badge don't match up. For example, I was downloading all 158 chapters of Kaguya-sama, but the download badge says I only have 114 downloaded chapters and 118 unread chapters, both of which aren't correct. When I click on the manga, it shows all 158 chapters and says each are downloaded.

 1. Select a manga with many chapters, e.g. Kaguya-sama from MangaDex
 2. Choose download all
 3. Unread and downloaded badges don't match

**Comment by Silwyn** [comment]
If two chapters have the same name, as sometime the case at MangaDex due to by different groups, then only chapter gets downloaded. Then all chapters of that title get marked as downloaded but the download counter only increases by one. If you then choose any of those chapters you will always get the downloaded version, even if it would differ. 

You might argue that this should not be the expected behaviour, but thats how it is. I assume this is due to how downloaded chapters are referenced in the storage. In other words chapters with the same name will reference the same folder and therefore are considered downloaded.

TLDR: Number of downloaded chapters correct due to all chapters of the same name being declared downloaded if one of them has been downloaded.

**Comment by arkon** [comment]
It should be more consistent post-#3513, but it may still be an issue in cases where things get orphaned.
